### ISSUE TITLE
Authenticated Paths for Proxy Provider

### ISSUE THREAD

**Issue by benedikt-bartscher** [goauthentik/authentik#5925]
**Is your feature request related to a problem? Please describe.**
Currently we can only configure "Unauthenticated Paths" for a Proxy Provider. This setting allow multiple regexes to check the URL against.
However there is no way to only protect specific paths.

**Describe the solution you'd like**
Add "Authenticated Paths" option to Proxy Providers or use another regex library which supports negative lookahead.

**Describe alternatives you've considered**
I tried regex negative lookaheads but authentik outpost are writting in golang, which does not support this.

**Additional context**
-

**Comment by q20** [comment]
Agreed. I attempted to allow Unauthenticated Paths for all paths _excluding_ `/admin/`, but it seems there is no way to achieve this. A simple _not_ function would be useful.  👍

**Comment by Rami-Pastrami** [comment]
I attempted this using a negative look-ahead regex (?!) but that does not seem to work, not sure if this is intentional or a bug. However the method, having a way to have authenticated paths would be useful
